src/common/common.go

```diff
@@ -132,3 +132,36 @@ func IsK8sNodeIsReady(node v1.Node) bool {
 	}
 	return false
 }
+
+// BuildHostsMapIPAddressBased builds a map containing all the IP addresses of the nodes in the
+// inventory so that later we can match reporting nodes based on the IP and not only on the name.
+func BuildHostsMapIPAddressBased(inventoryHostsMap map[string]inventory_client.HostData) map[string]inventory_client.HostData {
+	knownIpAddresses := map[string]inventory_client.HostData{}
+	for _, v := range inventoryHostsMap {
+		for _, ip := range v.IPs {
+			knownIpAddresses[ip] = v
+		}
+	}
+
+	return knownIpAddresses
+}
+
+// Matching of the node happens based on 2 rules
+//   * if the name of the node and in the inventory is exactly the same, use use it
+//   * if the name is not known in the inventory, we check if the IP address of the
+//     reporting node is known to the inventory
+// Using those rules we can cover the cases where e.g. inventory expects a short
+// hostname, but the node reports itself using its FQDN
+func HostMatchByNameOrIPAddress(node v1.Node, namesMap, IPAddressMap map[string]inventory_client.HostData) (inventory_client.HostData, bool) {
```

Comment:
I don't think there's the same potential problem with this one, but it would still be good to have unit tests.

Reply:
Tests created